### ISSUE TITLE
Undo screensaver with volume keys

### ISSUE THREAD

**Issue by Obegg** [Akemi/mpv-oled-screensaver#1]
I want to be able to undo the black screensaver with my volume keys the same way the script detects that I move my cursor

**Comment by Akemi** [comment]
are those keys even recognised by mpv or are those global multimedia keys? you can test with `mpv --input-test --force-window --idle`. if it's the latter, and the keys are not recognised by mpv, it's not possible.

**Comment by Akemi** [comment]
should any other keys also remove the screensaver? like all the keys that change brightness, colour, etc? should it be done on any key press?

**Comment by Obegg** [comment]
> are those keys even recognised by mpv or are those global multimedia keys? you can test with `mpv --input-test --force-window --idle`. if it's the latter, and the keys are not recognised by mpv, it's not possible.

recognized by mpv? what does it mean? pretty confusing wording but I use this:
```
UP	 	add volume +1
DOWN 		add volume -1
WHEEL_UP	add volume +1
WHEEL_DOWN	add volume -1
```
I hope it answers your question

> should any other keys also remove the screensaver? like all the keys that change brightness, colour, etc? should it be done on any key press?

Uh, I didn't think of that to be honest, that's a great idea (even tho I removed any keys that change those things, so it's pretty useless to me)

**Comment by Akemi** [comment]
yeah that answers the question. though this might not be possible if not catching or reacting on all key presses. first of all, mpv doesn't have an api that can tell the script which key is bound to which function. in that case one can only react on the default key binding, which wouldn't help your case.

the second limitation, i believe mpv's scripting api doesn't have a keybinding chaining mechanism. so one keybinding can only have one function bound to it. if i want to react on WHEEL_UP i would overwrite your binding instead.

i believe this can't be implemented in a satisfactory way atm.

**Comment by Obegg** [comment]
What about your suggestion?

>
> should it be done on any key press?

If it detects any key press (regardless of keybinding) = undo screensaver

**Comment by Akemi** [comment]
same, one would need to override any keybinding.

**Comment by Obegg** [comment]
Should I close this issue or keep it open?

**Comment by Akemi** [comment]
keep it open for now, i will look into it when i have some free time and maybe we can work around the limitation.
